lisa/tools/python.py

```diff
@@ -49,17 +49,30 @@ def _install(self) -> bool:
         self.node.os.install_packages(package_name)
         return self._check_exists()
 
-    def install_packages(self, packages_name: str) -> None:
-        cmd_result = self.run(
-            f"install -q {packages_name}",
-        )
-        if 0 != cmd_result.exit_code and get_matched_str(
-            cmd_result.stdout, self._no_permission_pattern
-        ):
+    def install_packages(self, packages_name: str, cache_dir="") -> None:
+        if (cache_dir == ""):
             cmd_result = self.run(
                 f"install -q {packages_name}",
-                sudo=True,
             )
+        else:
+            cmd_result = self.run(
+                f"install -q {packages_name} --cache-dir={cache_dir} -b {cache_dir}",
```

Comment:
Compose the command before run it. So it won't appear multiple times, like,
```python
cmd = f"install -q {packages_name}"
if cache_dir:
  cmd += f" --cache-dir={cache_dir} -b {cache_dir}"
```

Reply:
changed